quartz/processors/parse.ts

```diff
@@ -24,14 +26,19 @@ export function createProcessor(ctx: BuildCtx): QuartzProcessor {
       .use(remarkParse)
       // MD AST -> MD AST transforms
       .use(
-        transformers
-          .filter((p) => p.markdownPlugins)
-          .flatMap((plugin) => plugin.markdownPlugins!(ctx)),
-      )
+        transformers.flatMap((plugin) => plugin.markdownPlugins?.(ctx) ?? []),
+      ) as unknown as QuartzMdProcessor
+  ) // todo ^ figure out why the types dont align
+}
+
+export function createHtmlProcessor(ctx: BuildCtx): QuartzHtmlProcessor {
+  const transformers = ctx.cfg.plugins.transformers
+  return (
+    unified()
       // MD AST -> HTML AST
       .use(remarkRehype, { allowDangerousHtml: true })
       // HTML AST -> HTML AST transforms
-      .use(transformers.filter((p) => p.htmlPlugins).flatMap((plugin) => plugin.htmlPlugins!(ctx)))
+      .use(transformers.flatMap((plugin) => plugin.htmlPlugins?.(ctx) ?? []))
```

Comment:
how does this work when setting with `--conccurency` > 1?

iirc unified parser are singleton? (i might be wrong here)

Reply:
The unist processor itself works non-concurrently if that's what you're asking.

Each worker creates and instance of this processor independently.

Again, my change of `filter(..).flatMap(..)` to just `flatMap(..)` is perfectly equivalent here, there's no change in logic - it was a minor syntax change I barely paid any mind to :sweat_smile: 

Just in case the syntax sugar in the flatmap expands to this:
```ts
list.flatMap((plugin) => {
  if (plugin.htmlPlugins) {
    return plugin.htmlPlugins(ctx)  
  } else {
    return [] // flatMap(..) is basically map(..).flat(), so you can see how returning an empty list here is the same as filtering
  }
})
```